test/mitmproxy/test_optmanager.py

```diff
@@ -469,3 +477,44 @@ def test_set():
     opts.process_deferred()
     assert "deferredsequenceoption" not in opts.deferred
     assert opts.deferredsequenceoption == ["a", "b"]
+
+
+def test_load_paths(tdata):
+    opts = TS()
+    conf_path = tdata.path("mitmproxy/data/test_config.yml")
+    optmanager.load_paths(opts, conf_path)
+    assert opts.scripts == [
+        str(Path.home().absolute().joinpath("abc")),
+        str(Path(conf_path).parent.joinpath("abc")),
+        str(Path(conf_path).parent.parent.joinpath("abc")),
+        str(Path("/abc").absolute()),
+    ]
+    assert opts.not_scripts == ["~/abc", "abc", "../abc", "/abc"]
+
+
+@pytest.mark.parametrize(
+    "script_path, relative_to, expected",
+    (
+        ("~/abc", ".", Path.home().joinpath("abc")),
+        ("/abc", ".", Path("/abc")),
+        ("abc", ".", Path(".").joinpath("abc")),
+        ("../abc", ".", Path(".").joinpath("../abc")),
+        ("~/abc", "/tmp", Path.home().joinpath("abc")),
+        ("/abc", "/tmp", Path("/abc")),
+        ("abc", "/tmp", Path("/tmp").joinpath("abc")),
+        # Edge case for MacOS
```

Comment:
What's the edge case here?

(Let's have a short explanation in the doscstring)